src/metpy/calc/thermo.py

```diff
@@ -292,13 +299,25 @@ def moist_lapse(pressure, temperature, reference_pressure=None):
 
     """
     def dt(t, p):
-        t = units.Quantity(t, temperature.units)
-        p = units.Quantity(p, pressure.units)
-        rs = saturation_mixing_ratio(p, t)
-        frac = ((mpconsts.Rd * t + mpconsts.Lv * rs)
-                / (mpconsts.Cp_d + (mpconsts.Lv * mpconsts.Lv * rs * mpconsts.epsilon
-                                    / (mpconsts.Rd * t * t)))).to('kelvin')
-        return (frac / p).magnitude
+        partial_press = (
+            sat_pressure_0c_in_base_units
+            * np.exp(17.67 * (t - 273.15) / (t - 29.65))
+        )
+        rs = (
+            epsilon_in_base_units * partial_press
+            / (p - partial_press)
+        )
+        frac = (
+            (Rd_in_base_units * t + Lv_in_base_units * rs)
```

Comment:
😭

Reply:
Though this looks less awful without all the comments from the linter interspersed, so maybe it's not *as* bad.